schemas/constructs/component/v1beta1/component.json

```diff
@@ -0,0 +1,124 @@
+{
+	"$id": "https://schemas.meshery.io/component.json",
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"description": "Components are the atomic units for designing infrastructure. Learn more at https://docs.meshery.io/concepts/components",
+	"additionalProperties": false,
+	"type": "object",
+	"properties": {
+		"schemaVersion": {
+			"type": "string",
+			"description": "Specifies the version of the schema used for the definition."
+		},
+		"definitionVersion": {
+			"type": "string",
+			"description": "Specifies the version of the definition."
+		},
+		"displayName": {
+			"$ref": "https://schemas.meshery.io/core.json#inputString",
+			"description": "Display name of the component."
+		},
+		"format": {
+			"type": "string",
+			"enum": [
+				"JSON",
+				"CUE"
+			],
+			"default": "JSON",
+			"description": "Format specifies the format used in the `schema` field. JSON will be used as a default format."
+		},
+		"model": {
+			"$ref": "https://schemas.meshery.io/v1alpha1/model.json#",
+			"description": "Model of the component. Learn more at https://docs.meshery.io/concepts/models"
+		},
+		"metadata": {
+			"type": "object",
+			"description": "Metadata contains additional information associated with the component.",
+			"required": [
+				"name",
+				"version"
+			],
+			"properties": {
+				"description": {
+					"type": "string",
+					"description": "Description of the component."
+				},
+				"capabilities": {
+					"type": "object",
+					"description": "Meshery manages components in accordance with their specific capabilities. This field explicitly identifies those capabilities largely by what actions a given component supports; e.g. metric-scrape, sub-interface, and so on. This field is extensible. ComponentDefinitions may define a broad array of capabilities, which are in-turn dynamically interpretted by Meshery for full lifecycle management."
+				},
+				"genealogy": {
+					"type": "string"
+				},
+				"isAnnotation": {
+					"type": "boolean",
+					"description": "Identifies whether the component should be treated as deployable entity or just for logical representation."
+				},
+				"shape": {
+					"type": "string"
+				},
+				"primaryColor": {
+					"type": "string"
+				},
+				"secondaryColor": {
+					"type": "string"
+				},
+				"svgWhite": {
+					"type": "string"
+				},
+				"svgColor": {
+					"type": "string"
+				},
+				"svgComplete": {
+					"type": "string"
+				},
+				"published": {
+					"type": "boolean",
```

Comment:
This allows for fine-grain control to register components selectively.
`published` is also a property in models, when `published` is marked as `false` for models it takes precedence, and the entire model (with all its entities) will not be registered.

Reply:
This is a good behavioral note. I have included this in the description. Although, I wonder whether the interplay between `status` will be confusing and whether `published` should be reprsented as a status instead.